index.js

```diff
@@ -15,122 +16,203 @@ function loader (registryOrVersion) {
       this.metadata = metadata == null ? 0 : metadata
       this.nbt = nbt || null
 
+      // Probably add a new feature to mcdata, e.g itemsCanHaveStackId
+      if (registry.type === 'bedrock') {
+        this.stackId = stackId ?? Item.nextStackId()
+      } else {
+        this.stackId = null
+      }
+
       const itemEnum = registry.items[type]
       if (itemEnum) {
         this.name = itemEnum.name
         this.displayName = itemEnum.displayName
         if ('variations' in itemEnum) {
-          for (const i in itemEnum.variations) {
-            if (itemEnum.variations[i].metadata === metadata) { this.displayName = itemEnum.variations[i].displayName }
-          }
+          const variation = itemEnum.variations.find((item) => item.metadata === metadata)
+          if (variation) this.displayName = variation.displayName
         }
         this.stackSize = itemEnum.stackSize
+        // The 'itemEnum.maxDurability' checks to see if this item can lose durability
+        if (itemEnum.maxDurability && !this.durabilityUsed) this.durabilityUsed = 0
       } else {
         this.name = 'unknown'
         this.displayName = 'unknown'
         this.stackSize = 1
       }
     }
 
-    static equal (item1, item2, matchStackSize = true) {
+    static equal (item1, item2, matchStackSize = true, matchNbt = true) {
       if (item1 == null && item2 == null) {
         return true
       } else if (item1 == null) {
         return false
       } else if (item2 == null) {
         return false
       } else {
-        return (item1.type === item2.type &&
-            (matchStackSize ? item1.count === item2.count : true) &&
-            item1.metadata === item2.metadata &&
-            JSON.stringify(item1.nbt) === JSON.stringify(item2.nbt))
+        return (
+          item1.type === item2.type &&
+          (matchStackSize ? item1.count === item2.count : true) &&
+          item1.metadata === item2.metadata &&
+          (matchNbt ? JSON.stringify(item1.nbt) === JSON.stringify(item2.nbt) : true)
+        )
       }
     }
 
-    static toNotch (item) {
+    // Stack ID
+    // Probably needs to be moved to prismarine-registry later on as calling the loader again will reset it
+    static currentStackId = 0
+    static nextStackId () {
+      return Item.currentStackId++
+    }
+
+    static toNotch (item, serverAuthoritative = true) {
       if (registry.supportFeature('itemSerializationAllowsPresent')) {
```

Comment:
This function is hard to read as you do a feature check for registry type on bedrock but not pc.  You can simplify this function with 

```js
static toNotch(item, serverAuthoritative = true) {
  const hasNBT = item.nbt && item.nbt.value.length > 0
  if (registry.type === 'pc') {
    if (registry.supportFeature('itemSerializationAllowsPresent')) {
      if (item == null) return { present: false }
      return {
        present: true,
        itemId: item.type,
        itemCount: item.count,
        nbtData: item.nbt
      }
    } else if (registry.supportFeature('itemSerializationUsesBlockId')) {
      if (item == null) return { blockId: -1 }
      return {
        blockId: item.type,
        itemCount: item.count,
        itemDamage: item.metadata,
        nbtData: item.nbt
      }
    }
  } else if (registry.type === 'bedrock') {
    if (item == null || item.type === 0) return { network_id: 0 }
    if (registry.supportFeature('itemSerializeUsesAuxValue')) {
      return {
        network_id: item.id,
        auxiliary_value: (item.metadata << 8) | (item.count & 0xff),
        can_place_on: item.blocksCanPlaceOn,
        can_destroy: item.blocksCanDestroy,
        blocking_tick: 0,
        has_nbt: hasNBT,
        nbt: hasNBT ? { version: 1, nbt: item.nbt } : undefined
      }
    } else {
      return {
        network_id: item.type,
        count: item.count,
        metadata: item.metadata,
        has_stack_id: serverAuthoritative,
        stack_id: serverAuthoritative ? item.stackId : undefined,
        block_runtime_id: 0,
        extra: {
          can_place_on: item.blocksCanPlaceOn,
          can_destroy: item.blocksCanDestroy,
          blocking_tick: 0,
          has_nbt: hasNBT,
          nbt: hasNBT ? { version: 1, nbt: item.nbt } : undefined
        }
      }
    }
  }
  throw new Error("Don't know how to serialize for this mc version ")
}
```